ui/src/app/app-routing.module.ts

```diff
@@ -32,233 +31,242 @@ import {environment} from 'environments/environment';
-                  },
-                  resolve: {
-                    conceptSet: ConceptSetResolver,
-                  }
-                }]
-              }]
+                resolve: {
+                  cohort: CohortResolver,
+                }
+              }
+            ]
+          },
+          {
+            path: 'concepts',
+            component: ConceptHomepageComponent,
+            data: {
+              title: 'Search Concepts',
+              breadcrumb: {
+                value: 'Concepts',
+                intermediate: true
+              }
+            }
+          },
+          {
+            path: 'concepts/sets',
+            data: {
+              breadcrumb: {
+                value: 'Concept Sets',
+                intermediate: true
+              }
+            },
+            children: [{
+              path: '',
+              component: ConceptSetListComponent,
+              data: {
+                title: 'View Concept Sets',
+              }
+            }, {
+              path: ':csid',
+              component: ConceptSetDetailsComponent,
+              data: {
+                title: 'Concept Set',
+                breadcrumb: {
+                  value: 'Param: Concept Set Name',
+                }
+              },
+              resolve: {
+                conceptSet: ConceptSetResolver,
+              }
+            }]
           }]
-      },
-      {
-        path: 'admin/review-workspace',
-        component: AdminReviewWorkspaceComponent,
-        data: {title: 'Review Workspaces'}
-      }, {
-        path: 'admin/review-id-verification',
-        component: AdminReviewIdVerificationComponent,
-        data: {title: 'Review ID Verifications'}
-      }, {
-        path: 'profile',
-        component: ProfilePageComponent,
-        data: {title: 'Profile'}
-      }, {
-        path: 'settings',
-        component: SettingsComponent,
-        data: {title: 'Settings'}
-      }, {
-        path: 'workspaces/build',
-        component: WorkspaceEditComponent,
-        data: {title: 'Create Workspace', mode: WorkspaceEditMode.Create}
-      }
-    ]
+      }]
+  },
+  {
+    path: 'admin/review-workspace',
+    component: AdminReviewWorkspaceComponent,
+    data: {title: 'Review Workspaces'}
+  }, {
+    path: 'admin/review-id-verification',
+    component: AdminReviewIdVerificationComponent,
+    data: {title: 'Review ID Verifications'}
+  }, {
+    path: 'profile',
+    component: ProfilePageComponent,
+    data: {title: 'Profile'}
+  }, {
+    path: 'settings',
+    component: SettingsComponent,
+    data: {title: 'Settings'}
+  }, {
+    path: 'workspaces/build',
+    component: WorkspaceEditComponent,
+    data: {title: 'Create Workspace', mode: WorkspaceEditMode.Create}
   }
 ];
 
+fp.forEach(route => {
+  if (!route.isPublic) {
+    route.canActivate = [SignInGuard];
+    route.canActivateChild = [SignInGuard, RegistrationGuard];
+    route.runGuardsAndResolvers = 'always';
```

Comment:
I've messed around with TS annotations a bit and I'm surprised that something like this would work (but it might, since it compiled). I would recommend running locally with `--aot` here and manually testing to double check: https://github.com/all-of-us/workbench/blob/3862f270ed7f3360d7ce97a41dd741ffac7fecf2/ui/docker-compose.yaml#L12

Reply:
This totally doesn't work with `--aot` :( 
May I pair with you at some point? I can't figure out a good way to do this that doesn't add ridiculous amounts of boilerplate.

Reply:
Otherwise... I don't know, maybe we actually do punt on this PR and make the router conversion PR a bit bigger. I think this is much more straightforward in react/with the terra router.